From 2d31bbd969fa84d3ca2aaee44060b8eb2729e83f Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Mon, 17 Feb 2020 19:19:07 +0800
Subject: [PATCH 23/37] MLK-23331 arm64: dts: imx8mn: Update settings according
 to latest datasheet

commit 3b56409a1fd7385046460772d22368b0c6fdec62 from
https://source.codeaurora.org/external/imx/linux-imx.git

According to latest datasheet Rev.0.1, 03/2020, VDD_ARM does
NOT have dependency on VDD_SOC, so below table in datasheet
can be used directly for VDD_ARM:

Clock		Voltage
1.2GHz		0.85V
1.4GHz		0.95V
1.5GHz		1.0V

For DDR4 EVK board, system runs at nominal mode, so GPU can
ONLY run up to 400MHz.

For LPDDR4 EVK board, system runs at over-drive mode, so GPU
can run up to 600MHz.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Jacky Bai <ping.bai@nxp.com>
(cherry picked from commit e619dfe015d96f9322b59f386e00167ec1aab321)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/freescale/fsl-imx8mn-ddr4-evk.dts     | 19 +++++++++++++------
 1 file changed, 13 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mn-ddr4-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mn-ddr4-evk.dts
index 3e44a615b214..a3dad86bff7a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mn-ddr4-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mn-ddr4-evk.dts
@@ -960,12 +960,6 @@
 };
 
 &A53_0 {
-	operating-points = <
-		/* kHz    uV */
-		1500000 1000000
-		1400000 950000
-		1200000 950000
-	>;
 	arm-supply = <&buck2_reg>;
 };
 
@@ -978,6 +972,19 @@
 };
 
 &gpu {
+	assigned-clocks = <&clk IMX8MN_CLK_GPU_CORE_SRC>,
+			<&clk IMX8MN_CLK_GPU_SHADER_SRC>,
+			<&clk IMX8MN_CLK_GPU_AXI>,
+			<&clk IMX8MN_CLK_GPU_AHB>,
+			<&clk IMX8MN_GPU_PLL>,
+			<&clk IMX8MN_CLK_GPU_CORE_DIV>,
+			<&clk IMX8MN_CLK_GPU_SHADER_DIV>;
+	assigned-clock-parents = <&clk IMX8MN_GPU_PLL_OUT>,
+				<&clk IMX8MN_GPU_PLL_OUT>,
+				<&clk IMX8MN_SYS_PLL1_800M>,
+				<&clk IMX8MN_SYS_PLL1_800M>;
+	assigned-clock-rates = <0>, <0>, <800000000>, <400000000>, <1200000000>,
+				<400000000>, <400000000>;
 	status = "okay";
 };
 
-- 
2.17.1

